p2p/src/transport.rs

```diff
@@ -0,0 +1,22 @@
+#[derive(Debug, thiserror::Error)]
+pub enum Error {}
+
+pub trait Connection: Clone {
+    type Stream;
+
+    fn open(&self) -> Result<Self::Stream, Error>;
+}
+
+pub trait Endpoint {
```

Comment:
why the endpoint initiates the connection? shouldn't it represent a IP/port or other remote/local address?

Reply:
It does have a listen_addr, it represents the local binding returned by bind, so a working socket if you will.